ttt-game/src/components/board.js

```diff
@@ -0,0 +1,102 @@
+import React, {useReducer, useEffect, useContext} from 'react'
+import PlayerOne from './player-one.js'
+import PlayerTwo from './player-two.js'
+import TimeMachine from './TimeMachine.js'
+import { ScoreContext } from '../contexts/ScoreContext.js'
+
+const board = [null,null,null,null,null,null,null,null,null]
+let turn = 0
+let timeSteps = []
+
+
+  
+const Board = () => {
+  const [score, setScore] =  useContext(ScoreContext)
+  const boardReducer = (state, action) => {
+    switch (action.type) {
+      case 'move':
+        return state.map((box, index) => {
+          if(index === action.index && turn % 2 !== 0) return "X"
+          if(index === action.index && turn % 2 === 0 ) return "O"
+          return box
+        })
+        
+      case 'back':
+        return action.step
+        
+      case 'gameOver':
+        return board
+        
+      default:
+        return state
+    }
+  }
+  const [state, dispatch] = useReducer(boardReducer, board)
```

Comment:
👍 👍 for `useReducer` in this case - I think this is a really good choice for storing the state. This lets you handle the time travel relatively easily.

Reply:
You could probably even let your `timeSteps` state be a part of this as well. That way, you could handle that with separate actions instead of with useEffect. You could even check for the winner each time the board is updated here in the reducer.